### DISCUSSION TITLE
Buckt not found 404

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#28104] [Questions] [answered]
https://github.com/user-attachments/assets/7e4c3512-2bbe-4b9b-b38d-58cff799f8d6

Related to - https://github.com/orgs/supabase/discussions/1761

For search engines:
```
{"statusCode":"404","error":"Bucket not found","message":"Bucket not found"}
```

**Comment by GaryAustin1** [comment]
Your buckets are not public, but your URL you are using has `public` as the first part of the path.  That is only valid for a public bucket.  Otherwise it thinks your bucket name is public...
![image](https://github.com/user-attachments/assets/eac283df-e13b-4225-8eab-fc71ae5ee6a2)

**Reply by nicitaacom** [reply]
https://ocjqrvyxxdmlzdiikbww.supabase.co/storage/v1/object/tickets-images/image.png_1721478947540
{"statusCode":"400","error":"Error","message":"headers must have required property 'authorization'"}


For context: I jsut want to let any user to upload image to bucket
Then I set publicUrl in table 'messages' and then I SELECT 'messages.image' with that publicUrl (I select it with supbaseAdmin)

**Reply by GaryAustin1** [reply]
That is because  you have to have an authorization header to meet RLS for a private bucket.  Or you need a signed URL.

**Reply by GaryAustin1** [reply]
Make your bucket a public bucket if you want to allow access to all files and use a publicUrl without headers or getting a signedUrl with the path each time.

**Reply by nicitaacom** [reply]
I made my bucket public

{"statusCode":"404","error":"Bucket not found","message":"Bucket not found"}

How to fix that?

**Reply by GaryAustin1** [reply]
If you have a public bucket then the result you get from getPublicUrl should work.  That function does not look at the server so has no idea what your bucket is or if it is even there.  It just builds a string from the information you give it.

Then if the bucket is public you can use that url string in an image tag, or browser call without an authorization header or meeting RLS to get the file.

As far as you getting the error you need to show the call you are making when you get the error.

**Reply by nicitaacom** [reply, marked as answer]
I fixed that
[image](https://ocjqrvyxxdmlzdiikbww.supabase.co/storage/v1/object/public/tickets-images/image.png_1721481909070)



https://github.com/user-attachments/assets/99c79ac7-b2ab-4ab8-833b-26a7a5c219ed


Thank you @GaryAustin1
